### PR TITLE
Implement mem read

### PR DESCRIPTION
**Pull request by n1tram1**

### CODE REVIEW

**Review by CohenArthur — APPROVED** [review]
:fire: :fire: :fire: :fire:

**Comment by n1tram1** [comment]
> fire fire fire fire

c'est du vrai fire cette fois ci ? pas de truc a changer ?

File: rust-module/mfrc522_inner/spi.rs [hunk 1]
```diff
@@ -125,6 +134,14 @@ impl Mfrc522Spi {
         Ok(())
     }
 
+    pub fn fifo_flush(dev: &mut SpiDevice) -> KernelResult {
+        let flush_byte: u8 = 1u8 << 7;
```
Comment (Skallwar) [inline rust-module/mfrc522_inner/spi.rs:138]: constify this like so: https://github.com/ks0n/mfrc522-linux/blob/085c8fe48cea6cf87103bd108c80d562bc25f974/rust-module/mfrc522_inner/command.rs#L1

**Review by Skallwar — CHANGES_REQUESTED** [review]
Could you also add ``spi_dev`` to mem_write ? That will make a more uniform interface for all commands

Don't forget to rustfmt if you did not do it

File: rust-module/mfrc522_inner/spi.rs (hunk 1 above)
Comment (CohenArthur) [reply rust-module/mfrc522_inner/spi.rs:138]: > Could you also add `spi_dev` to mem_write ? That will make a more uniform interface for all commands

I think we can do this later, I opened an issue already #57. This way we can merge this and keep going as this is mostly "maintenance"

File: rust-module/mfrc522_inner/spi.rs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 use kernel::spi::{Spi, SpiDevice};
 use kernel::{pr_info, Error, KernelResult};
 
-use super::{Mfrc522CommandByte, Mfrc522Command, Mfrc522PowerDown, Mfrc522Receiver};
+use super::{Mfrc522Command, Mfrc522CommandByte, Mfrc522PowerDown, Mfrc522Receiver};
+
+const FIFO_LEVEL_REG_FLUSH_SHIFT: u8 = 7;
```
Comment (CohenArthur) [inline rust-module/mfrc522_inner/spi.rs:6]: I think what @Skallwar meant is to put it INSIDE the implementation, like...

File: rust-module/mfrc522_inner/spi.rs [hunk 3]
```diff
@@ -135,7 +137,7 @@ impl Mfrc522Spi {
     }
```
Comment (CohenArthur) [inline rust-module/mfrc522_inner/spi.rs:138]:
```suggestion
const FIFO_LEVEL_REG_FLUSH_SHIFT: u8 = 7;
```
...so

**Review by CohenArthur — APPROVED** [review]

File: rust-module/mfrc522_inner/spi.rs (hunk 3 above)
Comment (CohenArthur) [reply rust-module/mfrc522_inner/spi.rs:138]: Since this is what we do in the `Mfrc522Command` impl

File: rust-module/mfrc522_inner/spi.rs (hunk 2 above)
Comment (Skallwar) [reply rust-module/mfrc522_inner/spi.rs:6]: At the end o we did not do that for the other one, so here is just fine for me

**Review by Skallwar — APPROVED** [review]

**Pull request merged**